### PR TITLE
11 add validation of variables

### PR DESCRIPTION
**Pull request by GustaafL**
- ssl is now the only setting to set the scheme and the ssl. 
- basic validation of an email address
- check if host has https, fail if it does
- tests with parameterize from issue #16

### CODE REVIEW

File: src/flexmeasures_client/client.py [hunk 1]
```diff
@@ -46,10 +47,20 @@ class FlexMeasuresClient:
     session: ClientSession | None = None
 
     def __post_init__(self):
-        if not self.scheme:
-            self.scheme: str = "http" if "localhost" in self.host else "https"
-        if self.ssl is None:
-            self.ssl: bool = False if "localhost" in self.host else True
+        if not re.match(r".+\@.+\..+", self.email):
+            print(self.email)
```
Comment (victorgarcia98) [inline src/flexmeasures_client/client.py:51]: I would remove this print.

File: src/flexmeasures_client/client.py [hunk 2]
```diff
@@ -46,10 +47,20 @@ class FlexMeasuresClient:
...
+        if self.api_version not in API_VERSIONS_LIST:
+            raise ValueError(f"version not in versions list: {API_VERSIONS_LIST}")
+        # if ssl then scheme is https.
+        if self.ssl:
+            self.scheme = "https"
+        else:
+            self.scheme = "http"
+        if re.match(r"http\:\/\/|https\:\/\/", self.host):
+            print(self.host)
+            raise ValueError("scheme is inferred from ssl and can not be passed in the host")
```
Comment (victorgarcia98) [inline src/flexmeasures_client/client.py:62]: In this case, I think we should try to remove the schema from the host and warn the user that it shouldn't pass the scheme within the host.

**Review by victorgarcia98 — CHANGES_REQUESTED** [review]
Good work, this validation will save us a lot of time! (e.g: me using the username instead of  the email for the field email :laughing: )

I think we should also include the following validations:
* `len(self.email) > MIN_EMAIL_LENGTH`
* `len(self.password) > MIN_PASSWORD_LENGTH`

PS: I like the tests :D

File: src/flexmeasures_client/client.py (hunk 1 above)
Comment (GustaafL) [reply src/flexmeasures_client/client.py:51]: Thanks for the review. removed the print. Logging will be a later issue #14

File: src/flexmeasures_client/client.py (hunk 2 above)
Comment (GustaafL) [reply src/flexmeasures_client/client.py]: I think it's better to let it just fail but show why it fails. I don't want dependencies to be created on incorrect usage of the client itself. If users of the FlexMeasures clients want to quality of life improvements like automatically stripping the scheme I think that should happen in their app. We could maybe add this in home assistant for instance.

File: src/flexmeasures_client/client.py [hunk 3]
```diff
@@ -22,6 +23,7 @@
 POLLING_TIMEOUT = 200.0  # seconds
 REQUEST_TIMEOUT = 20.0  # seconds
 POLLING_INTERVAL = 10.0  # seconds
+API_VERSIONS_LIST = ("v1", "v1_1", "v1_2", "v1_3", "v2_0", "v3_0")
```
Comment (Flix6x) [inline src/flexmeasures_client/client.py:26]: Just `("v3_0",)`, please. (The comma is needed to keep this a tuple.) The client lacks most functionality for supporting the older versions, plus: FlexMeasures just dropped support for versions 1 to 2.

**Review by Flix6x — CHANGES_REQUESTED** [review]

File: src/flexmeasures_client/client.py (hunk 3 above)
Comment (GustaafL) [reply src/flexmeasures_client/client.py]: Thanks, removed the versions that are not supported and fixed the tuple

File: tests/test_client.py [hunk 4]
```diff
@@ -67,10 +77,49 @@ def test__init__():
...
+            "not an email address format string",
+        ),
+        (
+            {
+                "host": "http://test",
+                "email": "test@test.test",
+                "password": "test_password",
+            },
+            ValueError,
+            "scheme is inferred from ssl and can not be passed in the host",
```
[the remark this answers is not in the document]
Comment (GustaafL) [reply tests/test_client.py]: Changed the error message to give more specific hints. Separated http and https for different errors.

**Review by Flix6x — APPROVED** [review]

**Pull request merged**